buildSrc/src/main/kotlin/io/spine/internal/gradle/report/coverage/FileExtensions.kt

```diff
@@ -0,0 +1,128 @@
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.gradle.report.coverage
+
+import io.spine.internal.gradle.report.coverage.ClassMarker.ANONYMOUS
+import io.spine.internal.gradle.report.coverage.FileExtension.COMPILED_CLASS
+import io.spine.internal.gradle.report.coverage.FileExtension.JAVA_SOURCE
+import io.spine.internal.gradle.report.coverage.PathMarker.GENERATED
+import io.spine.internal.gradle.report.coverage.PathMarker.GRPC_SRC_FOLDER
+import io.spine.internal.gradle.report.coverage.PathMarker.JAVA_OUTPUT_FOLDER
+import io.spine.internal.gradle.report.coverage.PathMarker.JAVA_SRC_FOLDER
+import io.spine.internal.gradle.report.coverage.PathMarker.SPINE_JAVA_SRC_FOLDER
+import java.io.File
+
+/**
+ * This file contains extension methods and properties for `java.io.File`.
+ */
+
+/**
+ * Parses the name of a class from the absolute path of this file.
+ *
+ * Treats the fragment between the [precedingMarker] and [extension] as the value to look for.
+ * In case the fragment is located and it contains `/` symbols, they are treated
+ * as Java package delimiters and are replaced by `.` symbols before returning the value.
+ *
+ * If the absolute path of this file has either no [precedingMarker] or no [extension],
+ * returns `null`.
+ */
+internal fun File.parseClassName(
+    precedingMarker: PathMarker,
+    extension: FileExtension
+): String? {
+    val index = this.absolutePath.lastIndexOf(precedingMarker.infix)
+    return if (index > 0) {
+        var inFolder = this.absolutePath.substring(index + precedingMarker.length)
+        if (inFolder.endsWith(extension.value)) {
+            inFolder = inFolder.substring(0, inFolder.length - extension.length)
+            inFolder.replace('/', '.')
+        } else {
+            null
+        }
+    } else {
+        null
+    }
+}
+
+/**
+ * If this file contains the name of a class — according to the passed [classNameParser]s —
+ * appends the specified [destination] with it.
+ */
+internal fun File.appendTo(
+    destination: MutableList<String>,
+    vararg classNameParser: (file: File) -> String?
+) {
```

Comment:
Let's change the signature:
```kotlin
internal fun File.matches(vararg matcher: (File) -> String?): List<String>
```

This way we see what's happening a bit better. It's strange to me that the result of the method is stored into a mutable list parameter.